lms/djangoapps/instructor/views/api.py

```diff
@@ -1717,6 +1721,61 @@ def get_student_enrollment_status(request, course_id):
     return JsonResponse(response_payload)
 
 
+class StudentProgressUrlSerializer(serializers.Serializer):
+    """Serializer for course renders"""
+    unique_student_identifier = serializers.CharField(write_only=True)
+    course_id = CourseKeyField(required=False)
```

Comment:
why `required` is set to `False` whereas course_id is needed to get progress url?

Reply:
Its coming from URL as `kwarg`